dnf-repoconfig-daemon/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(SERVER_BIN dnf-repoconfig-daemon)
+set(SERVER_SRCS
+    main.cpp
+    repoconf.cpp
+    configuration.cpp
+)
+
+find_package(sdbus-c++ REQUIRED)
```

Comment:
Is there a C++ dbus library you can use that _doesn't_ require libsystemd? This would be a problem for distributions like Yocto and PCLOS, which don't have libsystemd.

Reply:
This is also problematic for using this in Midipix, because libsystemd does not build on musl libc.

Reply:
Oh cool, I wasn't aware that there's an sdbus-c++ :grin: 

The only other realistic choices would be gdbus (with glibmm, or just using its C API) and the classic libdbus. From my own experience, sdbus is by far the easiest and most robust library; gdbus is okay, and libdbus is .. rather involved.

Another option may be to write this in Python with dbus-python. The daemon is absolutely uncritical wrt. performance, and the ease of maintenance might be worth it. But I'm not a dnf maintainer, so I can't speak to whether any of these are realistic.

Reply:
Distros without systemd could just not build the daemon, right? It's not crucial. It is mostly aimed at GNOME software and Cockpit anyway, and Cockpit absolutely requires systemd, and GNOME runs rather poorly without it.

Reply:
@martinpitt The existing `dnfdaemon` doesn't require systemd, and there are distros adopting DNF without systemd. Don't suggest screwing them over.

Reply:
@martinpitt The goal would be that the existing `dnfdaemon` would be replaced, so `dnfdragora` would use it too. But if the new daemon doesn't work everywhere the old one does, I'm stuck with supporting both in `dnfdragora`. And that's no bueno.

Reply:
@Conan-Kudo : Ack, thanks for the background!